exercises.ipynb

```diff
@@ -172,22 +456,430 @@
+       " 'RUB': 71.0786,\n",
+       " 'HRK': 7.46,\n",
+       " 'JPY': 120.43,\n",
+       " 'THB': 33.623,\n",
+       " 'CHF': 1.1013,\n",
+       " 'SGD': 1.5129,\n",
+       " 'PLN': 4.2535,\n",
+       " 'BGN': 1.9558,\n",
+       " 'TRY': 6.3761,\n",
+       " 'CNY': 7.844,\n",
+       " 'NOK': 10.1638,\n",
+       " 'NZD': 1.7326,\n",
+       " 'ZAR': 16.828,\n",
+       " 'USD': 1.1139,\n",
+       " 'MXN': 21.3164,\n",
+       " 'ILS': 3.9272,\n",
+       " 'GBP': 0.86008,\n",
+       " 'KRW': 1300.09,\n",
+       " 'MYR': 4.64}"
+      ]
+     },
+     "execution_count": 46,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "rates"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def to_euro(rates, curr, amount):\n",
+    "    \"\"\"Function that searchs for the specified currency in an exchange rate dictionary \n",
+    "    and converts it into euros.\n",
+    "    \n",
+    "    Args:\n",
+    "        rates (dict): A dictionary containing the exchange rates of several currencies to euro\n",
+    "        curr (str): Abbreviation of the currency to be converted.\n",
+    "        amount (float): Total amount of the currency to be converted.\n",
+    "\n",
+    "    Returns:\n",
+    "        A float with the converted amounted into euros. None if the currency was not found.\n",
+    "    \"\"\"\n",
+    "    \n",
+    "    try:\n",
+    "        exchanged = amount/rates[curr]\n",
+    "    except KeyError:\n",
+    "        print('Currency not found, returned a missing value')\n",
+    "        exchanged = None\n",
+    "\n",
+    "    return(exchanged)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def load_data(path):\n",
+    "    \"\"\"Function that loads a CSV file from the specified path as a Pandas DataFrame\n",
+    "    and adds to it a new column (value_euros) with the value in euros of the assets\n",
+    "    found in the CSV. Information on exchange rates is taken from the following API:\n",
+    "    https://api.exchangeratesapi.io\n",
+    "    \n",
+    "    Args:\n",
+    "        path (str): A path that specifies the location of a CSV file.\n",
+    "\n",
+    "    Returns:\n",
+    "        A Pandas DataFrame object with a new column (value_euros).\n",
+    "    \"\"\"\n",
+    "    \n",
+    "    #Loading the CSV file from the path as a Pandas DataFrame\n",
+    "    df = pd.read_csv(path)\n",
+    "    \n",
+    "    #Setting up connection with the API and extracting the exchange rates data\n",
+    "    res = requests.get('https://api.exchangeratesapi.io/latest')\n",
+    "    content = json.loads(res.content)\n",
+    "    rates = content['rates']\n",
+    "    \n",
+    "    #Iterating over the elements of the DataFrame\n",
+    "    exchanged = []\n",
+    "    for i in range(len(df)):\n",
```

Comment:
Iterating over the elements of a list to make a new list... sounds like a map, no?